azure/README.md

```diff
@@ -1,79 +1,276 @@
+# export PKR_VAR_vm_size="REPLACE_ME"
+# export PKR_VAR_ssh_username="REPLACE_ME"
+
+export CLOUD_PROVIDER=azure
+make image
 ```
 
-- Create a custom Azure VM image based on Ubuntu 20.04 having kata-agent and other dependencies.
-	[setting up authenticated registry support](../docs/registries-authentication.md)
+Use the output of the above command to populate the following environment variable it will be used while deploying the cloud-api-adaptor:
+
 ```bash
-export VM_SIZE="REPLACE_ME"
-cd image
-CLOUD_PROVIDER=azure make image && cd -
+# e.g. format: /subscriptions/.../resourceGroups/.../providers/Microsoft.Compute/images/...
+export AZURE_IMAGE_ID="REPLACE_ME"
+```
+
+## Build CAA Container Image
+
+Run the following set of commands from the root of this repository:
+
+```bash
+export registry="REPLACE_ME" # e.g. quay.io/username
+export CLOUD_PROVIDER=azure
+make image
+```
+
+## Deploy Kubernetes using AKS
+
+Make changes to the following environment variable as you see fit:
+
+```bash
+export CLUSTER_NAME="REPLACE_ME"
+export AKS_WORKER_USER_NAME="azuser"
+export SSH_KEY="~/.ssh/id_rsa.pub"
 ```
 
-The output image id will be used while running the cloud-api-adaptor, which get's uploaded to your Azure portal using Packer.
+Deploy AKS with single worker node to the same resource group we created earlier:
+
+```bash
+az aks create \
+    --resource-group "${RESOURCE_GROUP}" \
+    --name "${CLUSTER_NAME}" \
+    --location "${LOCATION}" \
+    --node-count 1 \
+    --node-vm-size Standard_F4s_v2 \
+    --ssh-key-value "${SSH_KEY}" \
+    --admin-username "${AKS_WORKER_USER_NAME}" \
+    --os-sku Ubuntu
+```
 
-- Export your Azure VM information and run k8s on it
+Download kubeconfig locally to access the cluster using `kubectl`:
 
 ```bash
-VM_NAME="REPLACE_ME"
-PEER_POD_NAME="OUTPUT_FROM_ABOVE"
+az aks get-credentials \
+    --resource-group "${RESOURCE_GROUP}" \
+    --name "${CLUSTER_NAME}"
 ```
 
-# Running cloud-api-adaptor
+Label the nodes so that CAA can be deployed on it:
 
-- If using Calico CNI, [configure](https://projectcalico.docs.tigera.io/networking/vxlan-ipip#configure-vxlan-encapsulation-for-all-inter-workload-traffic) VXLAN encapsulation for all inter workload traffic.
+```bash
+kubectl label nodes --all node-role.kubernetes.io/worker=
+```
+
+## Deploy Cloud API Adaptor
+
+> **NOTE**: If you are using Calico CNI on a different Kubernetes cluster, then, [configure](https://projectcalico.docs.tigera.io/networking/vxlan-ipip#configure-vxlan-encapsulation-for-all-inter-workload-traffic) VXLAN encapsulation for all inter workload traffic.
 
-- Create Service Principal for the CAA
+### AKS Resource Group permissions
+
+AKS deploys the actual resources like the worker nodes in another resource group that is named in the format `MC_<aks-resource-group-name>_<cluster-name>_<location>`. For the CAA to be able to create pod VM in the same subnet as the worker nodes of the AKS cluster, run the following command:
```

Comment:
`az aks show --resource-group $RG --name $AKS_NAME --query nodeResourceGroup -o tsv`  should give us the node rg. alternatively we can maybe specifiy the node rg explicity at cluster creation?

Reply:
Yep there is a flag that be used to specify the rg name.